plugins/ssr/ssripats.ml

```diff
@@ -133,6 +133,12 @@ let intro_clear ids future_ipats =
     isCLR_PUSHL clear_ids
 end
 
+let tacCHECK_HYPS_EXIST hyps = Goal.enter begin fun gl ->
+  let ctx = Goal.hyps gl in
+  List.iter (Ssrcommon.check_hyp_exists ctx) hyps;
```

Comment:
Are you sure raising an error is what we want? I forgot, what is the recommended way to have a tactic fail @ppedrot ?

Reply:
Ok, I don't see why this exception catching is correlated to focusing on a goal, but I should get a tactic monad crash course before asking questions.

Reply:
For historical reasons, `Goal.enter` was made to catch exceptions. This was done in order to ease the porting of non-monadic code, as a typical pattern would have been going from
```
fun gl -> blah
```
to
```
enter begin fun gl -> blah end
```
where `blah` would typically raise exceptions to signal failure. This pattern is still used a lot in the current code, even though it's a bit fragile, as one needs to be careful not to add closures in the surrounding context of exception raise.